src/rest/connection.hpp

```diff
@@ -0,0 +1,135 @@
+/*
+ *  Copyright (c) 2020, The OpenThread Authors.
+ *  All rights reserved.
+ *
+ *  Redistribution and use in source and binary forms, with or without
+ *  modification, are permitted provided that the following conditions are met:
+ *  1. Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *  2. Redistributions in binary form must reproduce the above copyright
+ *     notice, this list of conditions and the following disclaimer in the
+ *     documentation and/or other materials provided with the distribution.
+ *  3. Neither the name of the copyright holder nor the
+ *     names of its contributors may be used to endorse or promote products
+ *     derived from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ *  AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ *  IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ *  ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ *  LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ *  CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ *  SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ *  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ *  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ *  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/**
+ * @file
+ *   This file includes connection definition for RESTful HTTP server.
+ */
+
+#ifndef OTBR_REST_CONNECTION_HPP_
+#define OTBR_REST_CONNECTION_HPP_
+
+#include <chrono>
+#include <unordered_map>
+#include <unordered_set>
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+
+#include "http_parser.h"
+
+#include "rest/request.hpp"
+#include "rest/resource.hpp"
+#include "rest/response.hpp"
+
+using std::chrono::steady_clock;
+
+namespace otbr {
+namespace rest {
+
+class Connection
+{
+public:
+    Connection(steady_clock::time_point aStartTime, Resource *aResource, int aFd);
+
```

Comment:
I think if we have the `Disconnect` function, it is  like current `SetComplete` function which closes fd and tells rest server "I need to be erased" ,then server could erase this connection in next loop. but as we discuss this morning,  the `complete` definition is more proper towards a request instead of a connection. So overall,  I think `Disconnect` looks more meaningful, but seems should take the same functionality as `SetComplete`. Is it right?